docs: update README for new orb (#403)
cypress-io/circleci-orb: README.md

See the official [CircleCI documentation](https://circleci.com/docs/2.0/using-orbs/).

---

In CircleCi, a "job" is just a collection of steps to carry out an action and a "command" defines a sequence of steps as a map to be executed in a job."

Below are the jobs and commands that will allow you to run your Cypress tests with an out-of-the-box or customized configuration.

## Jobs

### run

A single, complete job to run Cypress tests in your application. This is the out-of-the-box solution that will work for most use-cases.

## Basic Setup

A typical project can have:

```yaml
version: 2.1
orbs:
  # "cypress-io/cypress@3" installs the latest published
  # version "s.x.y" of the orb. We recommend you then use
  # the strict explicit version "cypress-io/cypress@3.x.y"
  # to lock the version and prevent unexpected CI changes
  cypress: cypress-io/cypress@3
workflows:
  build:
    jobs:
      - cypress/run # "run" job comes from "Cypress" orb
```

By using the `run` job definitions that Cypress provides inside the orb, it
brings simplicity and static checks of parameters to your CircleCI
configuration.

You can find multiple examples at

> [our official orb page](https://circleci.com/developer/orbs/orb/cypress-io/cypress)

Reviewer: ```suggestion
[our official orb page](https://circleci.com/developer/orbs/orb/cypress-io/cypress).
```

### Arguments

You can pass arguments to the `cypress/run` job to override any default behaviors. You can find the full list of arguments at [https://circleci.com/developer/orbs/orb/cypress-io/cypress#jobs-run](https://circleci.com/developer/orbs/orb/cypress-io/cypress#jobs-run)

## Parallelization

A more complex project that needs to install dependencies, build an application
and run tests across 4 CI machines [in parallel](/guides/guides/parallelization)
may have:

> > Note: `--parallel` flag tells Cypress to run tests in parallel. `parallelism` parameter tells Circle to run across `n` number of machines.`

Reviewer: ```suggestion
> Note: `--parallel` flag tells Cypress to run tests in parallel. `parallelism` parameter tells Circle to run across `n` number of machines.
```

> # Cypress CircleCI Orb

Reviewer: I think we lost some important structure here from the last current version. Like breaking these sections up with titles, like **Usage**, etc.

The [Cypress CircleCI Orb](https://github.com/cypress-io/circleci-orb) is a
piece of configuration set in your `.circleci/config.yml` file to correctly

> install, cache and run Cypress with very little effort.

Reviewer: Feels like we should give a quick blurb here about what a Circle Jobs & Commands is just to set the stage a little for this doc. Even as simple as...

"In CircleCi, a "job" is just a collection of steps to carry out an action and a "command" defines a sequence of steps as a map to be executed in a job."

Full documentation can be found at the

> [`cypress-io/circleci-orb`](https://github.com/cypress-io/circleci-orb) repo.

Reviewer: Isn't this where we already are?

For the Orb Quick Start Guide and usage cases, view the CircleCI
[Cypress orb documentation](https://circleci.com/developer/orbs/orb/cypress-io/cypress).

---

> 

Reviewer: Feels like we should give a quick blurb here about what a Circle Jobs & Commands is just to set the stage a little for this doc. Even as simple as...

"In CircleCi, a "job" is just a collection of steps to carry out an action and a "command" defines a sequence of steps as a map to be executed in a job." 

Below are the jobs and commands that will allow you to run your Cypress tests with an-out-of-the-box or customized configuration.

Reviewer: Otherwise it seems a lot of the other popular Orbs just link to the Circle registry listing in their readme's to have one source of truth.

> ## Jobs

Reviewer: Feels like we should give a quick blurb here about what a Circle Job is just to set the stage a little for this doc. Even as simple as...

"In CircleCi, a job is just a collection of steps to carry out an action."

### run

> A single, complete job to run Cypress tests in your application

Reviewer: ```suggestion
A single, complete job to run Cypress tests in your application.
```

> ## Basic Setup

Reviewer: We should add something indicating this is the out-of-the-box solution mentioned above.

A typical project can have:

```yaml
version: 2.1
orbs:
  # "cypress-io/cypress@3" installs the latest published
  # version "s.x.y" of the orb. We recommend you then use
  # the strict explicit version "cypress-io/cypress@3.x.y"
  # to lock the version and prevent unexpected CI changes
  cypress: cypress-io/cypress@3
workflows:
  build:
    jobs:
      - cypress/run # "run" job comes from "cypress" orb
```

By using the `run` job definitions that Cypress provides inside the orb, it
brings simplicity and static checks of parameters to your CircleCI
configuration.

> You can find multiple examples at

Reviewer: This is probably going away, right? Or will there still be examples on the Circle registry?

Author: No the examples from the `src/examples` folder will be displayed on this page after we publish

Author: See this deploy preview https://circleci.com/developer/orbs/orb/cypress-io/cypress?version=dev:4136de7f3dbd28ba722c91118588231c27d567a6#usage-examples

[our official orb page](https://circleci.com/developer/orbs/orb/cypress-io/cypress)

> ### Arguments

Reviewer: We should indicate why you'd pass additional params as part of this section.

**install-command** **_(String)_**

Overrides the default NPM command (npm ci)

**post-install** **_(String)_**

Additional command(s) to run after running install but before verifying Cypress

> and saving cache.

Reviewer: ```suggestion
and caching.
```

**cypress-cache-key** **_(String)_**

Cache key used to cache the Cypress binary.

> The default value is `cypress-cache-{{ arch }}-{{ checksum: "package.json" }}`

**cypress-cache-path** **_(String)_**

By, default, this will cache the `~/.cache/Cypress` directory so that the

> Cypress binary is cached. You can override this by providing your own cache path

Reviewer: ```suggestion
Cypress binary is cached. You can override this by providing your own cache path.
```

**node-cache-version** **_(String)_**

> Used to change the default node cache version (v1) if you need to clear the

Reviewer: ```suggestion
Used to change the default node cache version (v1) if you need to clear this
```

cache for any reason.

**include-branch-in-node-cache-key** **_(String)_**

> If true, the node cache will only apply to jobs within the same branch. Turned

Reviewer: The meaning of jobs within the same branch isn't super clear. Also maybe we reword the second sentence to something like, "The default is set to false".

off by default

**working-directory** **_(String)_**

Directory containing the `package.json` file. Used primarily for those using

> monorepos.

Reviewer: Feel like good context. 

```suggestion
monorepos or your cypress tests aren't at the root of the repository.
```

**package-manager** **_(Enum)_**

Select the default node package manager to use. NPM v5+ required.

> Available values include: `npm` (default), `yarn`, or `yarn-berry`

**start-command** **_(String)_**

> Command used to start your local dev server for Cypress to test against

Reviewer: ```suggestion
Command used to start your local dev server for Cypress to test against.
```

> **cypress-commandy** **_(String)_**

Reviewer: ```suggestion
**cypress-command** **_(String)_**
```

> Command used to run your Cypress tests including any flags you want to use

Reviewer: ```suggestion
Command used to run your Cypress tests, including any flags you want to use.
```

**parallelism** **_(Integer)_**

Number of Circle machines to use for load balancing, min 1 (requires `parallel`

> and `record` flags in your `cypress-command`). Defaults to 1

Reviewer: ```suggestion
and `record` flags in your `cypress-command`). Defaults to 1.
```

**install-browsers** **_(Boolean)_**

> Flag used to install additional browsers to run your tests. This is only needed

Reviewer: We may want to clarify that Cypress uses Electron by default, so if you need other browsers...

Reviewer: May also be a good time to clarify this is using https://circleci.com/developer/orbs/orb/circleci/browser-tools so it's clear why edge isn't included, etc

if you are passing the `--browser` flag in your `cypress-command`. This will
install the Chrome and Firefox binaries at runtime. If you need additional
browser support (example: `edge`), you can leave this flag as false and make
sure you use an executor with a docker image that includes the browsers of your
choosing. See
[https://hub.docker.com/r/cypress/browsers/tags](https://hub.docker.com/r/cypress/browsers/tags).

## Parallelization

A more complex project that needs to install dependencies, build an application
and run tests across 4 CI machines [in parallel](/guides/guides/parallelization)

> may have:

Reviewer: ```suggestion
may have:
Note: `--parallel` flag tells Cypress to run tests in parallel. `parallelism` parameter tells Circle to run across `n` number of machines.`
```

```yaml
version: 2.1
orbs:
  cypress: cypress-io/cypress@3
workflows:
  build:
    jobs:
      - cypress/run:
          cypress-command: "npx cypress run --record --parallel"
          start-command: "npm start"
          parallelism: 4
```

### Consumption

There are 2 key metrics to understand when running a CI job across multiple

> machines, consumption time and the actual time it takes. Consumption time is

Reviewer: ```suggestion
machines, **consumption time** and the actual time it takes **for tests to run**. Consumption time is
```

essentually the amount of CircleCI resources that a job requires to execute. For

> example you may have a job that runs on 5 machines and takes 1 minute for all to

Reviewer: ```suggestion
example, you may have a job that runs on 5 machines and takes 1 minute for all to
```

> complete. In this example it would only take 1 minute of time to execute all the

Reviewer: ```suggestion
complete. In this example it would only take 1 minute of actual time to execute all the
```

jobs but would consume 5 minutes of CircleCI resources. The Cypress CircleCI Orb
was designed to be as simple and fast as possible for the majority of use cases.
This means that if you are running your tests in parallel across more than 5

> machines then you may not want to use the `cypress/run` job as it will consume

Reviewer: ```suggestion
machines, then you may not want to use the `cypress/run` job as it will consume
```

more CircleCI resources than are necessary.

> **Parallelization Across More Than 5 Machines**
>

> > To lower your consumption in when running in parallel on more than 5 machines,

Reviewer: ```suggestion
> To lower your consumption time when running in parallel on more than 5 machines,
```

> > please checkout [this example](#) using the `cypress/install` and

Reviewer: ```suggestion
> see [this example](#) using the `cypress/install` and
```

> > `cypress/run-tests` commands

Reviewer: ```suggestion
> `cypress/run-tests` commands.
```

>

> > Check out the Cypress CircleCI <Icon name="github" inline="true" callout="Example Using Multiple Jobs" url="#" />.

Reviewer: This can come out

## Commands

### install

> Command that installs your application's node modules and Cypress dependencies

Reviewer: ```suggestion
Command that installs your application's node modules and Cypress dependencies.
```

> #### Arguments

Reviewer: Should these args be duplicated here as well?

**install-command** **_(String)_**

> Overrides the default NPM command (npm ci)

Reviewer: ```suggestion
Overrides the default NPM command (npm ci).
```

**post-install** **_(String)_**

Additional command(s) to run after running install but before verifying Cypress

> and saving cache.

Reviewer: ```suggestion
and caching.
```

### Arguments

> You can pass additional arguments to the `cypress/run` job to override any default behaviors.

Reviewer: The args section is tough to take in and decipher as-is. Is there a way to get these into a table or something so it's easier to grok at a glance?
